subsystems/drivebase/SwerveDrive.java

```diff
@@ -311,8 +327,24 @@ public void mustangPeriodic() {
         Logger.recordOutput(DRIVEBASE_PITCH, getPitch());
         Logger.recordOutput(DRIVEBASE_ROLL, getRoll());
 
+       
+
+        if (kConfig.kDriveMotorType.equals(Motor_Type.KRAKEN_X60)) {
+            emaCurrent = currentAlpha * ((TalonFX) mModules[0].getDriveMotor()).getSupplyCurrent() + (1 - currentAlpha) * emaCurrent; // weighted current average of front-right drive motor
+            if (emaCurrent > 30) { // TODO tune this? what should be the threshold?
+                currentLimitConfigs.SupplyCurrentLimit = currentSupplyCurrentLimit - 0.25; // TODO what increment to decrease/increase limit? how frequently current readings update?
```

Comment:
This reducing by 0.25 and adding back 0.25 is just a bit too cute for my taste. No half measures, please. If we believe the current draw is a problem, drop it to a level that we think is safe, even if that level is going to be painful. If you think we need something like 10 amps continuous for the shooter and another 10 while the intake and indexer are running, that's probably something like 25 amps input per drive motor. Or even 20 (the steer motors need something, too). We can always force the condition on our test field and see how it behaves for real